src/main/ports.ts

```diff
@@ -83,13 +97,93 @@ export const getUnexpectedOpenPorts = (
   return configPorts.filter((port) => openPorts.includes(port));
 };
 
+export const assignPortsToNode = (node: Node): Node => {
+  // Retrieve all used ports and convert them to strings
+  const { p2pPorts: usedP2pPorts, otherPorts: usedOtherPorts } =
+    getPodmanPorts();
+  const usedPorts = [...usedP2pPorts, ...usedOtherPorts].map(
+    (port) => port?.toString() ?? '',
+  );
+
+  // Define the port types to update
+  const portTypes = [
+    'httpPort',
+    'p2pPorts',
+    'enginePort',
+    'p2pPortsUdp',
+    'p2pPortsTcp',
+    'webSocketsPort',
+    'quicPortUdp',
+  ]; // Add other relevant port types
+
+  const executionService = getNodePackageByServiceNodeId(
+    node.id,
+  )?.services.find((service) => {
+    return service.serviceId === 'executionClient';
+  });
+
+  const nodePackages = getNodePackages();
+  // Update ports using array methods
+  portTypes.forEach((portType) => {
+    const defaultPort =
+      node.spec.configTranslation?.[portType]?.defaultValue ?? null;
+    const currentPort = node.config.configValuesMap[portType];
+
+    if (!defaultPort && !currentPort) {
+      return;
+    }
+
+    // Use current port, or default if not initialized, converted to a number
+    let assignedPort = Number.parseInt(
+      (currentPort || defaultPort) as string,
+      10,
+    );
+
+    // Find next available port if the current/default one is in use
+    if (
+      nodePackages.length > 1 &&
+      usedPorts.includes(assignedPort.toString())
+    ) {
+      assignedPort = Number.parseInt(
+        findNextAvailablePort(usedPorts, assignedPort),
+        10,
+      );
+    }
+
+    // Add the newly assigned port to the used ports list
+    usedPorts.push(assignedPort.toString());
+    // Update the node configuration with the port as a string
+    node.config.configValuesMap[portType] = assignedPort.toString();
+  });
+
+  if (node.spec.rpcTranslation === 'eth-l1-beacon' && executionService) {
+    const executionNode = getNode(executionService.node.id);
+    let executionEndpoint = node.config.configValuesMap.executionEndpoint;
+
+    // Check if the endpoint is enclosed in quotes
+    const isQuoted =
+      executionEndpoint.startsWith('"') && executionEndpoint.endsWith('"');
+    const portSuffix = `:${executionNode.config.configValuesMap.enginePort}`;
+
+    // Append the port inside the quotes if necessary
+    if (isQuoted) {
```

Comment:
Ohhh yeah good catch. Hmm.. I think it would be beneficial to add a conditional like you've suggested, otherwise, this **will** temporarily break all existing nodes being run by users right now, unless they change the setting in the ports manually... I agree with NN being Alpha, and think it'd still be worth the small change to help others. Will look into this real quick, it shouldn't be difficult!